Assets/Documentation/man/man3/Glow.3
.TH "Glow" 3 "Version 0.1" "NPP VR Simulation" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Glow \- This class implements logic to apply a glow effect to an object\&.  

.SH SYNOPSIS
.br
.PP
.PP
Inherits MonoBehaviour\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "void \fBsetGlowing\fP ()"
.br
.RI "This method toggles the glow effect on and off\&. "
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "void \fBStart\fP ()"
.br
.RI "This method initialises the objectRenderer to render the glow effect and sets the strength of the effect\&. "
.ti -1c
.RI "IEnumerator \fBAnimateGlow\fP (float startValue, float endValue, float duration)"
.br
.RI "This method interpolates the glow strength\&. "
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "float \fBglowStrength_on\fP = 1\&.5f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIglowStrength_on\fP specifies the strength of the glow effect when turned on
.RE
.PP
"
.ti -1c
.RI "float \fBglowStrength_off\fP = 0\&.0f"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIglowStrength_off\fP specifies the strength of the glow effect when turned off
.RE
.PP
"
.ti -1c
.RI "bool \fBisGlowing\fP = false"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIisGlowing\fP checks if the glow effect is currently active active on the object 
.RE
.PP
"
.ti -1c
.RI "Renderer \fBobjectRenderer\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fIobjectRenderer\fP is a reference to a renderer object to render the glow effect 
.RE
.PP
"
.ti -1c
.RI "Material \fBmats\fP"
.br
.RI "
.PP
\fBParameters\fP
.RS 4
\fImats\fP is a Material being rendered by the renderer 
.RE
.PP
"
.in -1c
.SH "Detailed Description"
.PP 
This class implements logic to apply a glow effect to an object\&. 
.SH "Member Function Documentation"
.PP 
.SS "IEnumerator Glow\&.AnimateGlow (float startValue, float endValue, float duration)\fR [private]\fP"

.PP
This method interpolates the glow strength\&. 
.PP
\fBParameters\fP
.RS 4
\fIstartValue\fP lower bound for the interpolation
.br
\fIendValue\fP upper bound for the interpolation
.br
\fIduration\fP duration of the interpolation
.RE
.PP

.SS "void Glow\&.setGlowing ()"

.PP
This method toggles the glow effect on and off\&. 
.SS "void Glow\&.Start ()\fR [private]\fP"

.PP
This method initialises the objectRenderer to render the glow effect and sets the strength of the effect\&. 
.SH "Member Data Documentation"
.PP 
.SS "float Glow\&.glowStrength_off = 0\&.0f\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIglowStrength_off\fP specifies the strength of the glow effect when turned off
.RE
.PP

.SS "float Glow\&.glowStrength_on = 1\&.5f\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIglowStrength_on\fP specifies the strength of the glow effect when turned on
.RE
.PP

.SS "bool Glow\&.isGlowing = false\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIisGlowing\fP checks if the glow effect is currently active active on the object 
.RE
.PP

.SS "Material Glow\&.mats\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fImats\fP is a Material being rendered by the renderer 
.RE
.PP

.SS "Renderer Glow\&.objectRenderer\fR [private]\fP"

.PP

.PP
\fBParameters\fP
.RS 4
\fIobjectRenderer\fP is a reference to a renderer object to render the glow effect 
.RE
.PP


.SH "Author"
.PP 
Generated automatically by Doxygen for NPP VR Simulation from the source code\&.


Assets/Skripte/shader/scripts/Glow.cs
using System.Collections;
using UnityEngine;

/// <summary>
/// This class implements logic to apply a glow effect to an object.
/// </summary>
public class Glow : MonoBehaviour{


    /// <param name="glowStrength_on"> specifies the strength of the glow effect when turned on</param>
    [SerializeField] private float glowStrength_on = 1.5f;
    /// <param name="glowStrength_off"> specifies the strength of the glow effect when turned off</param>
    private float glowStrength_off = 0.0f;
    /// <param name="isGlowing"> checks if the glow effect is currently active active on the object </param>
    private bool isGlowing = false;
    /// <param name="objectRenderer"> is a reference to a renderer object to render the glow effect </param>
    private Renderer objectRenderer;
    /// <param name="mats"> is a Material being rendered by the renderer </param>
    private Material mats;

    /// <summary>
    /// This method initialises the objectRenderer to render the glow effect and sets the strength of the effect
    /// </summary>
    void Start()
    {
        if (objectRenderer == null)
        {
            objectRenderer = this.GetComponentInChildren<MeshRenderer>();
            Debug.Log("objectRenderer: " + objectRenderer);
        }

        mats = objectRenderer.material;
        mats.SetFloat("_GlowStrength", glowStrength_off);
    }

    /// <summary>
    /// This method toggles the glow effect on and off.
    /// </summary>
    public void setGlowing()
    {
        if (!isGlowing)
        {
            StartCoroutine(AnimateGlow(glowStrength_off, glowStrength_on, 1.0f)); // 1 second duration
            isGlowing = true;
        }
        else
        {
            StartCoroutine(AnimateGlow(glowStrength_on, glowStrength_off, 1.0f)); // 1 second duration
            isGlowing = false;
        }
    }

    /// <summary>
    /// This method interpolates the glow strength.
    /// </summary>
    /// <param name="startValue"> lower bound for the interpolation</param>
    /// <param name="endValue"> upper bound for the interpolation</param>
    /// <param name="duration"> duration of the interpolation</param>
    private IEnumerator AnimateGlow(float startValue, float endValue, float duration)
    {
        float elapsedTime = 0.0f;

        while (elapsedTime < duration)
        {
            float currentGlow = Mathf.Lerp(startValue, endValue, elapsedTime / duration);
            mats.SetFloat("_GlowStrength", currentGlow);
            elapsedTime += Time.deltaTime;
            yield return null;
        }

        mats.SetFloat("_GlowStrength", endValue);
    }
}